source/mir/algebraic.d

```diff
@@ -155,7 +155,8 @@ private static struct _Void()
 /++
 Checks if the type is instance of $(LREF Algebraic).
 +/
-enum bool isVariant(T) = is(immutable T == immutable Algebraic!Types, Types...);
+enum bool isVariant(T) = !is(immutable T == immutable noreturn)
+    && is(immutable T : immutable Algebraic!Types, Types...);
```

Comment:
the change is essentially changing the == to `:`, but also needs to check against noreturn.